### PR TITLE
Warn about potential incorrect use of cvmfs_talk

### LINKED ISSUE #3075: `cvmfs_talk` does not warn if `-i repo` is needed but not given
Commands in `cvmfs_talk` that need `-i repo` finish without any output. This creates confusion if the execution was successful or not. 

It would be better if `cvmfs_talk` gives feedback that arguments are missing.

### PR DESCRIPTION
**Pull request by jblomer**
Fixes #3075

### CODE REVIEW

File: cvmfs/cvmfs_talk.cc [hunk 1]
```diff
@@ -143,7 +143,8 @@ bool SendCommand(const std::string &command, InstanceInfo instance_info) {
 static void Usage(const std::string &exe) {
   LogCvmfs(kLogCvmfs, kLogStdout,
     "Usage: %s [-i instance | -p socket] <command>                     \n"
-    "   By default, iteratate through all instances.                   \n"
+    "   By default, iteratate through all instances defined in         \n"
```
Comment (vvolkl) [inline cvmfs/cvmfs_talk.cc]:
```suggestion
    "   By default, iterate through all instances defined in         \n"
```

File: cvmfs/cvmfs_talk.cc [hunk 2]
```diff
@@ -254,14 +255,23 @@ int main(int argc, char *argv[]) {
...
+      is_empty_repo_list = false;
       instance_info.instance_name = repos[i];
       LogCvmfs(kLogCvmfs, kLogStdout, "%s:", repos[i].c_str());
       bool retval = SendCommand(command, instance_info);
       if (!retval) retcode = 1;
     }
+    if (is_empty_repo_list) {
+      LogCvmfs(kLogCvmfs, kLogStdout,
+               "Warning: no instance was specified. In this case, the command "
+               "is executed for all instances defined in CVMFS_REPOSITORES but "
```
Comment (HereThereBeDragons) [inline cvmfs/cvmfs_talk.cc]: typo in CVMFS_REPOSITORIES

File: cvmfs/cvmfs_talk.cc [hunk 3]
```diff
@@ -254,14 +255,23 @@ int main(int argc, char *argv[]) {
...
+               "this list is empty. Did you mean running \n\n"
+               "    cvmfs_talk -i <repository name> <command>\n");
```
Comment (HereThereBeDragons) [inline cvmfs/cvmfs_talk.cc:273]: do you also want to add the option with `-p <socket>`?

**Review by HereThereBeDragons — CHANGES_REQUESTED** [review]
only a small but important typo

during testing i also found the following: 
```
cvmfs_talk
> Command missing
```

so maybe we can add that just calling cvmfs_talk falls back to `--help`? (could be another PR)

File: cvmfs/cvmfs_talk.cc (hunk 3 above)
Comment (jblomer) [reply cvmfs/cvmfs_talk.cc:273]: I think I'd rather not advertise the `-p` interface. Under normal circumstances, `-i` should be used.

**Review by vvolkl — APPROVED** [review]

**Pull request merged**